pandas/tests/reshape/test_concat.py

```diff
@@ -1279,6 +1279,33 @@ def test_concat_ignore_index(self, sort):
 
         tm.assert_frame_equal(v1, expected)
 
+    concat_index_names = [
+        (["idx", "idx"], "idx"),
+        (["idx", None], "idx"),
+        ([None, None], None),
+        (["idx1", "idx2"], None),
+        (["idx", "idx", None], "idx"),
+        (["idx1", "idx2", None], None),
+    ]
+
+    @pytest.mark.parametrize("input_names,output_name", concat_index_names)
+    def test_concat_same_index_names(self, input_names, output_name):
+        # GH13475
+        rows = ["a", "b", "c"]
+        cols = ["x", "y", "z"]
+        vals = [0, 1, 2]
+
+        indices = [pd.Index(rows, name=n) for n in input_names]
+        frames = [pd.DataFrame({c: vals}, index=i) for i, c in zip(indices, cols)]
+        result = pd.concat(frames, axis=1)
+
+        exp_ind = pd.Index(rows, name=output_name)
```

Comment:
nit pick: might want to define this inside the frames constructor rather than in a separate variable

Reply:
indeed, best to keep result / expected separate